src/components/common/ColonyHome/ColonySubscription/ColonySubscription.tsx

```diff
@@ -33,43 +39,86 @@ const MSG = defineMessages({
 });
 
 const ColonySubscription = () => {
-  const { colony } = useColonyContext();
-  const { colonyAddress } = colony || {};
+  const { colony, canInteractWithColony } = useColonyContext();
+  const {
+    user,
+    updateUser,
+    wallet,
+    walletConnecting,
+    connectWallet,
+    userLoading,
+  } = useAppContext();
+  const navigate = useNavigate();
 
-  const { user } = useAppContext();
+  const watchedItem = getWatchedColony(colony, user?.watchlist?.items);
 
-  const isSubscribed = !!(user?.watchlist?.items || []).find(
-    (item) => (item?.colony as Colony)?.colonyAddress === colonyAddress,
-  );
+  /* Watch (follow) a colony */
+  const [watch, { data: watchData, loading: loadingWatch }] =
+    useCreateWatchedColoniesMutation({
+      variables: {
+        input: {
+          colonyID: colony?.colonyAddress || '',
+          userID: user?.walletAddress || '',
+        },
+      },
+      refetchQueries: [
+        {
+          query: GetFullColonyByNameDocument,
+          variables: { name: colony?.name },
+        },
+      ],
+    });
+
+  /* Unwatch (unfollow) a colony */
+  const [unwatch, { data: unwatchData, loading: loadingUnwatch }] =
+    useDeleteWatchedColoniesMutation({
+      variables: { input: { id: watchedItem?.id || '' } },
+      refetchQueries: [
+        {
+          query: GetFullColonyByNameDocument,
+          variables: { name: colony?.name },
+        },
+      ],
+    });
+
+  /* Update user on watch/unwatch */
+  useEffect(() => {
+    if (updateUser) {
+      updateUser(user?.walletAddress);
+    }
+  }, [user, updateUser, watchData, unwatchData]);
+
+  const handleSubscribe = useCallback(() => {
+    if (user) {
+      watch();
+    } else if (wallet && !user) {
+      handleNewUser();
+      // TO Do: update to new user modal
+      navigate(CREATE_USER_ROUTE, { replace: true });
```

Comment:
Should we use replace here? We might want to allow people to go back to colony home from the user creation flow